drivers/gles3/storage/texture_storage.cpp

```diff
@@ -812,6 +812,10 @@ void TextureStorage::texture_2d_update(RID p_texture, const Ref<Image> &p_image,
 #endif
 }
 
+void TextureStorage::texture_2d_update_partial(RID p_texture, const Ref<Image> &p_sub_image, int p_dst_x, int p_dst_y, int p_dst_mip, int p_layer) {
```

Comment:
It would be preferable if the method took a sub-region of the source image, rather than forcing to first instantiate and copy it from the source image just to be passed to this function.